refnx/reflect/_mdsimulation.py

```diff
@@ -0,0 +1,307 @@
+        should be rotated through the xy-plane. This expects the neutrons or
+        X-ray to be incident at the bottom of the simulation cell (where
+        z=0).
+    roughness: float, optional
+        The fractional (of the layer thickness) roughness to be considered
+        between the layers in the simulation cell.
+    verbose: bool, optional
+        True if you want to be notified when the pdb and lgt files have been
+        read and when the sld profile is calculated.
+    structure: biopython.structure, attribute
+        The is the object which contains the information read in from the
+        simulation trajectory.
+    dimensions: float, array_like, attribute
+        An array of 3 floats containing the simulation cell dimensions.
+    """
+    def __init__(self, pdbfile, layer_thickness=1, cut_off=5, flip=False,
+                 roughness=0, verbose=False):
+        self.pdbfile = pdbfile
+        self.structure = None
+        self.dimensions = [0, 0, 0]
+        self.read_pdb()
+        self.verbose = verbose
+        if self.verbose:
+            print('PDB file read.')
+        self.av_layers = np.zeros((int(np.floor(self.dimensions[2] /
+                                                layer_thickness)) + 1, 5))
+        self.av_layers[:, 0] = layer_thickness
+        self.av_layers[:, 3] = layer_thickness * roughness
+        self.layers = np.array([self.av_layers, ] * len(self.structure))
+        self.flip = flip
+        self.cut_off = cut_off
+
+    def assign_scattering_lengths(self, lgtfile=None, radiation='neutron',
+                                  xray_energy=None, atom_types=None,
+                                  scattering_lengths=None):
+        """
+        Assign the scattering lengths either from a .lgt file, a user defined
+        array or allow it to be determined from the elements listed in the
+        .pdb file.
+        Parameters
+        ----------
+        radiation: str, optional
+            Either 'neutron' or 'xray'. This is the type of radiation that was
+            used in the experimental reflectometry measurements.
+        xray_energy: float, optional
+            Only required if the radiation is 'xray'. This is the energy of the
+            probing X-ray radiation, in units of kilo electron Volts
+        lgtfile: str, optional
+            The path and name of the .lgt file (if present), which contains the
+            scattering lengths of each of the atom type in the pdbfile.
+            Currently the .lgt file style that is supported is a 3 column space
+            separated txt file where the columns are atom_type,
+            real_scattering_length, and imaginary_scattering_length
+            respectively. If a lgtfile is not used the scattering lengths for
+            the system will be determined based on the element type that is
+            defined in the final column of the pdb file.
+        atom_types: str, array_like
```

Comment:
what is `atom_types` used for?

Reply:
the atom_types are paired with the scattering lengths. Basically they are the names corresponding to the appropriate scattering length (maybe it would be better done with a `dict`?)